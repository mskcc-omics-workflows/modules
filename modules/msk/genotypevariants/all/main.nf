process GENOTYPEVARIANTS_ALL {
    tag "$meta.id"
    label 'process_single'


    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'ghcr.io/msk-access/genotype_variants:sha-7e2c3c8a':
        'ghcr.io/msk-access/genotype_variants:sha-7e2c3c8a' }"

    input:
    tuple val(meta), path(bam_standard), path(bai_standard), path(bam_duplex), path(bai_duplex), path(bam_simplex), path(bai_simplex), path(maf)
    path(fasta)
    path(fai)

    output:
    tuple val(meta), path("*.maf"), emit: maf
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def sample = task.ext.prefix != null ? "-si ${task.ext.prefix}" : (meta.id != null ? "-si ${meta.id}" : "")
    def patient = meta.patient ?"-p ${meta.patient}": ''
    def bams_standard = bam_standard ?"-b $bam_standard" : ''
    def bam_liquid = (bam_duplex && bam_simplex) ? "-d $bam_duplex -s $bam_simplex" : ''

    """
    genotype_variants small_variants all \\
    -i ${maf} \\
    -r ${fasta} \\
    -g /usr/local/bin/GetBaseCountsMultiSample \\
    $patient \\
    $bams_standard \\
    $bam_liquid \\
    $sample \\
    -t $task.cpus \\
    $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        genotypevariants: \$(genotype_variants --version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def sample = "${meta.sample}"
    def patient = "${meta.patient}"

    """
    touch ${prefix}.maf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        genotypevariants: \$(genotype_variants --version)
    END_VERSIONS
    """
}
