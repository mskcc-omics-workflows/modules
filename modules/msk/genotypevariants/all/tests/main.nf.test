nextflow_process {

    name "Test Process GENOTYPEVARIANTS_ALL"
    script "../main.nf"
    process "GENOTYPEVARIANTS_ALL"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "genotypevariants"
    tag "genotypevariants/all"


    test("illumina - bam, msk - maf") {
        config './nextflow.config'
        when {
            process {
                """
                input[0] = [
                            [ id:'197', patient:'c-id'], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam_bai'], checkIfExists: true),
                            [],
                            [],
                            [],
                            [],
                            input[1] = file(params.test_data_mskcc['chr22']['chr22maf'], checkIfExists: true)
                        ]
                input[1] = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                input[2] = file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("illumina - duplex and simplex, msk - maf") {
        config './nextflow.config'
        when {
            process {
                """
                input[0] = [
                            [ id:'197', patient:'c-id'], // meta map
                            [],
                            [],
                            file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam_bai'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_paired_end_sorted_bam_bai'], checkIfExists: true),
                            file(params.test_data_mskcc['chr22']['chr22maf'], checkIfExists: true)
                        ]
                input[1] = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                input[2] = file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("illumina - bam - stub") {

        config './nextflow.config'
        options "-stub"

        when {
            process {
                """
                input[0] = [
                            [ id:'197', patient:'c-id'], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam_bai'], checkIfExists: true),
                            [],
                            [],
                            [],
                            [],
                            []
                        ]
                input[1] = []
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
