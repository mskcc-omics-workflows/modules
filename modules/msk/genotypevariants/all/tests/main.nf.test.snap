{
    "illumina - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "197",
                            "patient": "c-id"
                        },
                        "197.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,55fd807cb50fcc6611e6d352b86e4704"
                ],
                "maf": [
                    [
                        {
                            "id": "197",
                            "patient": "c-id"
                        },
                        "197.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,55fd807cb50fcc6611e6d352b86e4704"
                ]
            }
        ],
        "timestamp": "2024-06-18T14:18:45.350503"
    },
    "illumina - duplex and simplex, msk - maf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "197",
                            "patient": "c-id"
                        },
                        [
                            "197-DUPLEX_genotyped.maf:md5,6c0649cfd70ceb0587827f48bf93caf7",
                            "197-ORG-SIMPLEX-DUPLEX_genotyped.maf:md5,6f37d4e00ab92ce623bd286f6804f30a",
                            "197-SIMPLEX-DUPLEX_genotyped.maf:md5,6d6c71cdf73e74c10fcb68e477a4455f",
                            "197-SIMPLEX_genotyped.maf:md5,6c0649cfd70ceb0587827f48bf93caf7"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,55fd807cb50fcc6611e6d352b86e4704"
                ],
                "maf": [
                    [
                        {
                            "id": "197",
                            "patient": "c-id"
                        },
                        [
                            "197-DUPLEX_genotyped.maf:md5,6c0649cfd70ceb0587827f48bf93caf7",
                            "197-ORG-SIMPLEX-DUPLEX_genotyped.maf:md5,6f37d4e00ab92ce623bd286f6804f30a",
                            "197-SIMPLEX-DUPLEX_genotyped.maf:md5,6d6c71cdf73e74c10fcb68e477a4455f",
                            "197-SIMPLEX_genotyped.maf:md5,6c0649cfd70ceb0587827f48bf93caf7"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,55fd807cb50fcc6611e6d352b86e4704"
                ]
            }
        ],
        "timestamp": "2024-06-18T14:18:39.109687"
    },
    "illumina - bam, msk - maf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "197",
                            "patient": "c-id"
                        },
                        [
                            "197-ORG-STD_genotyped.maf:md5,8a140bbb7af1c291e8597866fb9d6bb9",
                            "197-STANDARD_genotyped.maf:md5,6c0649cfd70ceb0587827f48bf93caf7"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,55fd807cb50fcc6611e6d352b86e4704"
                ],
                "maf": [
                    [
                        {
                            "id": "197",
                            "patient": "c-id"
                        },
                        [
                            "197-ORG-STD_genotyped.maf:md5,8a140bbb7af1c291e8597866fb9d6bb9",
                            "197-STANDARD_genotyped.maf:md5,6c0649cfd70ceb0587827f48bf93caf7"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,55fd807cb50fcc6611e6d352b86e4704"
                ]
            }
        ],
        "timestamp": "2024-06-18T14:18:29.181063"
    }
}